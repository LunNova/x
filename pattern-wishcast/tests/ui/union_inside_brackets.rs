// SPDX-FileCopyrightText: 2025 LunNova
//
// SPDX-License-Identifier: MIT

//! Test that unsupported generic fields containing Self produce helpful errors

#![feature(never_type)]

use pattern_wishcast::pattern_wishcast;

pattern_wishcast! {
	enum IncorrectlyInsideBracket = { None };
	enum StuckEvaluation = {
		IncorrectlyInsideBracket |
		{
			Var { id: usize },
		}
	};
}

fn main() {}
