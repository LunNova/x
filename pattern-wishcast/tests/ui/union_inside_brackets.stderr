error: Union syntax (|) is not allowed inside braces. To compose types, use: enum MyEnum = TypeA | TypeB | { variants }
  --> tests/ui/union_inside_brackets.rs:12:28
   |
12 |         IncorrectlyInsideBracket |
   |                                  ^
